src/main/java/appeng/core/features/registries/MatterCannonAmmoRegistry.java

```diff
@@ -52,7 +52,7 @@ public float getPenetration( final ItemStack is )
 	{
 		for( final ItemStack o : this.DamageModifiers.keySet() )
 		{
-			if( Platform.isSameItem( o, is ) )
+			if( Platform.itemComparisons().isEqualItem( o, is ) )
 			{
 				return this.DamageModifiers.get( o ).floatValue();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) appeng.core.features.registries.MatterCannonAmmoRegistry.getPenetration(ItemStack) makes inefficient use of keySet iterator instead of entrySet iterator [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=findbugs%3AWMI_WRONG_MAP_ITERATOR)